site/publish.sh

```diff
@@ -47,15 +47,24 @@ function parallelfunction() {
         ;;
 
     azsync*)
+        # Load the env variables corresponding to use get-fileshare-signed-url.sh for the Azure File Share to sync, extracted from the end of the task name
+        envToLoad=".env-${1#azsync-}"
+        # shellcheck source=/dev/null
+        source "${envToLoad}"
+
+        # Required variables that should now be set from the .env file
+        : "${STORAGE_FILESHARE?}" "${JENKINS_INFRA_FILESHARE_CLIENT_ID?}" "${JENKINS_INFRA_FILESHARE_CLIENT_SECRET?}" "${FILESHARE_SYNC_SOURCE?}"
+
```

Comment:
```suggestion
        : "${STORAGE_FILESHARE?}" "${JENKINS_INFRA_FILESHARE_CLIENT_ID?}" "${JENKINS_INFRA_FILESHARE_CLIENT_SECRET?}" "${FILESHARE_SYNC_SOURCE?}"

# Ensure a trailing is always present (as it will be a source for a recursive `azcopy sync`)
FILESHARE_SYNC_SOURCE="${FILESHARE_SYNC_SOURCE%/}/"
```

To avoid any issue in the secrets (as the content is not visible)

Reply:
```suggestion
        : "${STORAGE_FILESHARE?}" "${JENKINS_INFRA_FILESHARE_CLIENT_ID?}" "${JENKINS_INFRA_FILESHARE_CLIENT_SECRET?}" "${FILESHARE_SYNC_SOURCE?}"

# Ensure a trailing slash is always present (as it will be a source for a recursive `azcopy sync`)
FILESHARE_SYNC_SOURCE="${FILESHARE_SYNC_SOURCE%/}/"
```
(nit)